core/src/main/java/org/elasticsearch/index/translog/Checkpoint.java

```diff
@@ -109,4 +158,50 @@ public int hashCode() {
         result = 31 * result + Long.hashCode(generation);
         return result;
     }
+
+    /** A copy of {@link OutputStreamIndexOutput} without the intermediate buffering */
+    static class DirectOutputStreamIndexOutput extends IndexOutput {
```

Comment:
Why not just use `OutputStreamIndexOutput`?  Is its buffering somehow problematic?